vdirsyncer/sync/__init__.py

```diff
@@ -241,7 +241,7 @@ def __init__(self, ident, dest):
 
     async def _run_impl(self, a, b):
         meta = self.dest.status.get_new(self.ident)
-        if not self.dest.storage.read_only:
+        if not self.dest.storage.read_only and not self.dest.storage.no_delete:
```

Comment:
I'd rather log this just to be clear. Also simplifies the `if` condition.

```suggestion
        if self.dest.storage.read_only or self.dest.storage.no_delete:
            sync_logger.debug(f"Skipping deletion of item {self.ident} from {self.dest.storage}")
        else:
```

Reply:
works fine for me.